crates/aide/src/helpers/no_api.rs

```diff
@@ -0,0 +1,102 @@
+use derive_more::{AsMut, AsRef, Deref, DerefMut, From};
+use serde::{Deserialize, Serialize};
+
+use crate::{OperationInput, OperationOutput};
+
+/// Allows non [OperationInput] or [OperationOutput] types to be used in aide handlers with a default empty documentation.  
+/// For types that already implement [OperationInput] or [OperationOutput] it overrides the documentation and hides it.
+/// ```ignore
+/// pub async fn my_sqlx_tx_endpoint(
+///     NoApi(mut tx): NoApi<Tx<sqlx::Any>> // allows usage of the TX
+/// ) -> NoApi<Json<YourResult>> // Hides the API of the return type
+/// # {}
+/// ```
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    Ord,
+    PartialOrd,
+    Eq,
+    PartialEq,
+    Hash,
+    Serialize,
+    Deserialize,
+    Deref,
+    DerefMut,
+    AsRef,
+    AsMut,
+    From,
+)]
+pub struct NoApi<T>(pub T);
```

Comment:
Couldn't this be simplified with something like below?
```rs
pub type NoApi<T> = WithApi<T, SomeTypeThatHasNoDocs>;
```

Reply:
Yes indeed

Reply:
Actually not, unformtunately rust requires the phantom data so
```
fn endpoint (NoApi(mut tx): NoApi<Tx<Postgres>>)
```
becomes
```
fn endpoint (NoApi(mut tx, _): NoApi<Tx<Postgres>>)
```

Reply:
Welp, yeah I see no way around that either, we could suggest using `.into_inner()` in downstream code, but the appearing extra field out of nowhere can be confusing indeed.

Reply:
The proper Dry code would be using a macro, but it's really not worth the horrible proc macro syntax for two impls, it's not like there is a lot of code duplication.

Reply:
maybe we could do something with the axum impls, they are quite similar... Maybe utility trait + macro ? but it's really complexifying things